src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -139,23 +151,53 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     //    This executable is our init-actions.sh, which will stand up our jupyter server and proxy.
     val initActions = Seq(new NodeInitializationAction().setExecutableFile(GcsPath(bucketName, GcsRelativePath(clusterResourcesConfig.initActionsScript)).toUri))
 
-    // Create a SoftwareConfig and set a property that makes the cluster have only one node
-    val softwareConfig = new SoftwareConfig().setProperties(Map("dataproc:dataproc.allow.zero.workers" -> "true").asJava)
+    val machineConfig = MachineConfig(clusterRequest.machineConfig, clusterDefaultsConfig)
+
+    // Create a config for the master node, if properties are not specified in request, use defaults
+    val masterConfig = new InstanceGroupConfig()
+      .setMachineTypeUri(machineConfig.masterMachineType.get)
+      .setDiskConfig(new DiskConfig().setBootDiskSizeGb(machineConfig.masterDiskSize.get))
 
     // Create a Cluster Config and give it the GceClusterConfig, the NodeInitializationAction and the SoftwareConfig
-    val clusterConfig = new ClusterConfig()
-      .setGceClusterConfig(gce)
-      .setInitializationActions(initActions.asJava).setSoftwareConfig(softwareConfig)
+    createClusterConfig(machineConfig).setGceClusterConfig(gceClusterConfig)
+      .setInitializationActions(initActions.asJava)
+      .setMasterConfig(masterConfig)
+  }
 
-    // Create a Cluster and give it a name and a Cluster Config
-    val cluster = new DataprocCluster()
-      .setClusterName(clusterName.string)
-      .setConfig(clusterConfig)
+  private def createClusterConfig(machineConfig: MachineConfig): ClusterConfig = {
+    // If the number of workers is zero, make a Single Node cluster, else make a Standard one
+    if (machineConfig.numberOfWorkers.get == 0) {
```

Comment:
Maybe note in the function comment that this function expects a fully specified (i.e. no `None`s) config.